### PR TITLE
feature(gatsby): Add experiment to run source plugins in parallel

### PR DESCRIPTION
**Pull request by KyleAMathews**
This speeds up cold sourcing for gatsbyjs.com (which has 4 external sources) from 78s -> 45s or a 42% improvement.

Gatsby runs APIs sequentially. That's normally what we want as running many APIs is CPU bound. But sourcing is almost entirely bounded by the speed of the external APIs so we save quite a bit of time for sites with multiple source plugins by sourcing in parallel.

There's a chance someone is depending on the serial behavior of API running so we'd need to wait for Gatsby v3 to enable it for everyone.

In the meantime you can enable it by adding the `PARALLEL_SOURCING` flag to your gatsby-config.js or running gatsby with the env variable `GATSBY_EXPERIMENTAL_PARALLEL_SOURCING=true`

### CODE REVIEW

File: packages/gatsby/src/utils/api-runner-node.js [hunk 1]
```diff
@@ -558,90 +558,109 @@ module.exports = async (api, args = {}, { pluginSource, activity } = {}) =>
       }
     }
 
-    Promise.mapSeries(implementingPlugins, plugin => {
-      if (stopQueuedApiRuns) {
-        return null
-      }
+    let apiRunPromiseOptions = {}
```
Comment (KyleAMathews) [inline packages/gatsby/src/utils/api-runner-node.js:561]: The PR has a lot less change than it appears with all the whitespace changes — see it with those removed https://github.com/gatsbyjs/gatsby/pull/28214/files?diff=unified&w=1

File: packages/gatsby/src/utils/api-runner-node.js [hunk 2]
```diff
@@ -558,90 +558,109 @@ module.exports = async (api, args = {}, { pluginSource, activity } = {}) =>
...
+    let runPromise
+    if (
+      process.env.GATSBY_EXPERIMENTAL_PARALLEL_SOURCING &&
+      api === `sourceNodes`
```
Comment (vladar) [inline packages/gatsby/src/utils/api-runner-node.js:565]: Can we swap those? Accessing `process.env` can be expensive: https://github.com/nodejs/node/issues/3104 

```suggestion
      api === `sourceNodes` &&
      process.env.GATSBY_EXPERIMENTAL_PARALLEL_SOURCING
```

**Review by vladar — CHANGES_REQUESTED** [review]
I think the PR is pretty safe - couldn't find a way how it can leak through the flag boundaries. But have one request for a change inline.

P.S. Curious to see the results of this experiment 👍

File: packages/gatsby/src/utils/api-runner-node.js (hunk 2 above)
Comment (KyleAMathews) [reply packages/gatsby/src/utils/api-runner-node.js]: wut? Who knew. Also so weird/unfortunate. Thanks!

**Review by vladar — APPROVED** [review]

**Pull request merged**